client/beefy/src/worker.rs

```diff
@@ -217,23 +218,28 @@ where
 			// TODO: (adoerr) Enacting a new authority set will also implicitly 'conclude'
 			// the currently active BEEFY voting round by starting a new one. This is
 			// temporary and needs to be replaced by proper round life cycle handling.
-			if active.id != self.rounds.validator_set_id() ||
-				(active.id == GENESIS_AUTHORITY_SET_ID && self.best_beefy_block.is_none())
+			if self.rounds.is_none() ||
```

Comment:
Note that this has potentially unintended consequence of setting `notification.header.number()` as the best currently known BEEFY-finalized block. This is incorrect, we can't just assume that the first ever GRANDPA-finalized block that we see (right after the worker starts) is immediately BEEFY-finalized.

Reply:
I see what you mean, but how is current behavior different than before (changes in this PR)?

Before, this check `active.id() != self.rounds.validator_set_id()` was always true for first block with same end result, right?

Reply:
Ugh, oh, you're right. I thought that `rounds` were initialized with the current validator set which means this branch would only be triggered after we transition, but indeed it was initialized with `empty`, hence the first incoming block was being considered as the best beefy block.

So I guess it's fine to merge as is, but this should be addressed in a follow up PRs, cause it's clearly not desired. Maybe let's create an issue in the gadget repo (if there isn't one already) and add a `TODO` comment with an issue number/link to make sure we don't miss it in the future?

Reply:
yes, there is a `TODO (adoerr)` comment here already, but I will change it for a tracked GH issue.

Reply:
opened https://github.com/paritytech/grandpa-bridge-gadget/issues/366 and referenced it in code as well